kernel/Kconfig

```
@@ -862,25 +862,47 @@ config XIP
 
 menu "Security Options"
 
-config STACK_CANARIES
-	bool "Compiler stack canaries"
-	depends on ENTROPY_GENERATOR || TEST_RANDOM_GENERATOR
-	select NEED_LIBC_MEM_PARTITION if !STACK_CANARIES_TLS
+config NEED_STACK_CANARIES
+	bool
 	help
-	  This option enables compiler stack canaries.
+	  Hidden option to signal that software stack protection is enabled.
 
+choice
+	prompt "Stack canaries protection options"
+	optional
+	help
 	  If stack canaries are supported by the compiler, it will emit
 	  extra code that inserts a canary value into the stack frame when
 	  a function is entered and validates this value upon exit.
 	  Stack corruption (such as that caused by buffer overflow) results
 	  in a fatal error condition for the running entity.
-	  Enabling this option can result in a significant increase
-	  in footprint and an associated decrease in performance.
+	  Enabling this option, depending on the level chosen, can result in a
+	  significant increase in footprint and a corresponding decrease in performance.
 
 	  If stack canaries are not supported by the compiler an error
 	  will occur at build time.
 
-if STACK_CANARIES
+config STACK_CANARIES
+	bool "Compiler stack canaries for all functions"
+	depends on ENTROPY_GENERATOR || TEST_RANDOM_GENERATOR
+	select NEED_LIBC_MEM_PARTITION if !STACK_CANARIES_TLS
+	select NEED_STACK_CANARIES
+	help
+	  This option enables compiler stack canaries for all functions.
+
+config STACK_CANARIES_STRONG
+	bool "Compiler stack canaries in most functions"
```

related to other Kconfig choice entry prompt proposal:

```suggestion
	bool "Most functions"
```
Only a proposal, thus non-blocking .